docs/source/en/model_doc/seamless_m4t.md

```diff
@@ -0,0 +1,202 @@
+
+*What does it take to create the Babel Fish, a tool that can help individuals translate speech between any two languages? While recent breakthroughs in text-based models have pushed machine translation coverage beyond 200 languages, unified speech-to-speech translation models have yet to achieve similar strides. More specifically, conventional speech-to-speech translation systems rely on cascaded systems that perform translation progressively, putting high-performing unified systems out of reach. To address these gaps, we introduce SeamlessM4T, a single model that supports speech-to-speech translation, speech-to-text translation, text-to-speech translation, text-to-text translation, and automatic speech recognition for up to 100 languages. To build this, we used 1 million hours of open speech audio data to learn self-supervised speech representations with w2v-BERT 2.0. Subsequently, we created a multimodal corpus of automatically aligned speech translations. Filtered and combined with human-labeled and pseudo-labeled data, we developed the first multilingual system capable of translating from and into English for both speech and text. On FLEURS, SeamlessM4T sets a new standard for translations into multiple target languages, achieving an improvement of 20% BLEU over the previous SOTA in direct speech-to-text translation. Compared to strong cascaded models, SeamlessM4T improves the quality of into-English translation by 1.3 BLEU points in speech-to-text and by 2.6 ASR-BLEU points in speech-to-speech. Tested for robustness, our system performs better against background noises and speaker variations in speech-to-text tasks compared to the current SOTA model. Critically, we evaluated SeamlessM4T on gender bias and added toxicity to assess translation safety. Finally, all contributions in this work are open-sourced and accessible at https://github.com/facebookresearch/seamless_communication*
+
+## Usage
+
+First, load the processor and a checkpoint of the model:
+
+```python
+>>> from transformers import AutoProcessor, SeamlessM4TModel
+
+>>> processor = AutoProcessor.from_pretrained("ylacombe/hf-seamless-m4t-medium")
+>>> model = SeamlessM4TModel.from_pretrained("ylacombe/hf-seamless-m4t-medium")
+```
+
+You can seamlessly use this model on text or on audio, to generated either translated text or translated audio.
+
+### Speech
+
+You can easily generate translated speech with [`SeamlessM4TModel.generate`]. Here is an example showing how to generate speech from English to Russian.
+
+```python
+>>> inputs = processor(text = "Hello, my dog is cute", src_lang="eng", return_tensors="pt")
+
+>>> audio_array = model.generate(**inputs, tgt_lang="rus")
+>>> audio_array = audio_array[0].cpu().numpy().squeeze()
+```
+
+You can also translate directly from a speech waveform. Here is an example from Arabic to English:
+
+```python
+>>> from datasets import load_dataset
+
+>>> dataset = load_dataset("arabic_speech_corpus", split="test[0:1]")
```

Comment:
This is a 1.9GB dataset. Maybe we can just load one sample using streaming mode to circumvent a lengthy download?

```python
from datasets import load_dataset

dataset = load_dataset("arabic_speech_corpus", split="test", streaming=True)
audio_sample = next(iter(dataset))["audio"]

inputs = processor(audios=audio_sample["array"], sampling_rate=audio_sample["sampling_rate"], return_tensors="pt")
```